### ISSUE TITLE
"Delete spam messages immediately without moving them to the trash folder" doesn't delete Email in Spam Folder

### ISSUE THREAD

**Issue by markusb68** [thunderbird/thunderbird-android#6582]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.501

### Where did you get the app from?

Other

### Android version

13

### Device model

Pixel 5

### Steps to reproduce

"Delete spam messages immediately without moving them to the trash folder" doesn't delete Email in Spam Folder, just removes it in the K9 Spam Folder.

Go to Spam Folder.
Delete Spam Emails.
All looks OK.

Open email account in web browser or IMAPSize.
The Spam Emails are still in the Spam / Junk folder. K9 just don't show them any more.

### Expected behavior

I'd expected K9 to delete message and expunge the mailbox.

### Actual behavior

 The Spam Emails are still in the Spam / Junk folder. K9 just don't show them any more.

### Logs

No Logs needed. Simple to reproduce.

**Comment by cketti** [comment]
Can you record a debug log when trying to delete a message from the Spam folder? See https://github.com/thundernest/k-9/wiki/LoggingErrors

**Comment by markusb68** [comment]
Hello,
just to make it clear:
It does delete the spam Mail in K-9 Spam folder view.
But it doesn't delete the spam email on the IMAP server (in it's spam folder).
Thus the Spam Mail is still visible on other IMAP Clients like Thunderbird on a PC  or K-9 on an other Tablet.

Here is the log:
[k9mail-logs.txt](https://github.com/thundernest/k-9/files/10432887/k9mail-logs.txt)

**Comment by cketti** [comment]
Thanks for the log.

The log doesn't contain the commands sent to the server. It looks like you only enabled debug logging after deleting a spam message. However, the log still contained enough information to reproduce the issue. It'll be fixed in the next beta.

**Comment by markusb68** [comment]
Nope, I did enable it, delete a spam message and exported the log.
I did it again, just in case...

`01-17 12:44:40.821 26932  5356 E BLASTBufferQueue: [VRI[MessageList]#24](f:0,a:3) Faking releaseBufferCallback from transactionCompleteCallback
01-17 12:44:40.836 26932 26956 I MessagingController: Running command 'deleteMessages', seq = 1426 (background priority)
01-17 12:44:40.836 26932 26956 V NotificationController: Removing notification for message MessageReference(accountUuid=29599d51-79e1-421c-871e-721647bb687f, folderId=5, uid=9376)
01-17 12:44:40.838 26932  5356 E BLASTBufferQueue: [VRI[MessageList]#24](f:0,a:3) Faking releaseBufferCallback from transactionCompleteCallback
01-17 12:44:40.842 26932 26956 D MessagingController: Not moving deleted messages to local Trash folder. Removing local copies.
01-17 12:44:40.852 26932 26956 V LockableDatabase: LockableDatabase: Transaction ended, took 3 ms / com.fsck.k9.mailstore.LocalFolder.setFlags(LocalFolder.java:871)
01-17 12:44:40.854 26932  5356 E BLASTBufferQueue: [VRI[MessageList]#24](f:0,a:2) Faking releaseBufferCallback from transactionCompleteCallback
01-17 12:44:40.856 26932 26956 D MessagingController: Delete policy for account 29599d51-79e1-421c-871e-721647bb687f is ON_DELETE
01-17 12:44:40.856 26932 26956 E MessagingController: Error running command 'deleteMessages'
01-17 12:44:40.856 26932 26956 E MessagingController: java.lang.IllegalArgumentException: Required value was null.
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.Preconditions.requireValidUids(Preconditions.kt:8)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingControllerCommands$PendingMoveAndMarkAsRead.create(MessagingControllerCommands.java:77)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.queueMoveOrCopy(MessagingController.java:834)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.deleteMessagesSynchronous(MessagingController.java:2099)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.lambda$deleteMessages$23(MessagingController.java:1968)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.$r8$lambda$92mkWTQcVE9qE-bAgJdJX56iLkA(Unknown Source:0)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController$$ExternalSyntheticLambda19.run(Unknown Source:10)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.runInBackground(MessagingController.java:222)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController.access$000(MessagingController.java:108)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at com.fsck.k9.controller.MessagingController$1.run(MessagingController.java:163)
01-17 12:44:40.856 26932 26956 E MessagingController: 	at java.lang.Thread.run(Thread.java:1012)
01-17 12:44:40.872 26932  5356 E BLASTBufferQueue: [VRI[MessageList]#24](f:0,a:2) Faking releaseBufferCallback from transactionCompleteCallback
`
